doc-processor-gradle-plugin/src/main/kotlin/nl/jolanrensen/docProcessor/gradle/ProcessDocTask.kt

```diff
@@ -0,0 +1,187 @@
+package nl.jolanrensen.docProcessor.gradle
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.NamedDomainObjectContainer
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.artifacts.Dependency
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.FileCollection
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Classpath
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.OutputFiles
+import org.gradle.api.tasks.TaskAction
+import org.gradle.workers.WorkerExecutor
+import org.jetbrains.dokka.DokkaSourceSetID
+import org.jetbrains.dokka.gradle.GradleDokkaSourceSetBuilder
+import java.io.File
+import javax.inject.Inject
+
+/**
+ * Process doc task you can instantiate in your build.gradle(.kts) file.
+ * For example using [Project.creatingProcessDocTask].
+ */
+abstract class ProcessDocTask @Inject constructor(factory: ObjectFactory) : DefaultTask() {
+
+    /** Source root folders for preprocessing. This needs to be set! */
+    @get:InputFiles
+    val sources: ListProperty<File> = factory
+        .listProperty(File::class.java)
+
+    /**
+     * Set base directory which will be used for relative source paths.
+     * By default, it is '$projectDir'.
+     */
+    @get:Input
+    val baseDir: Property<File> = factory
+        .property(File::class.java)
+        .convention(project.projectDir)
+
+    /**
+     * Target folder to place the preprocessing results in.
+     */
+    @get:Input
+    val target: Property<File> = factory
+        .property(File::class.java)
+        .convention(File(project.buildDir, "docProcessor${File.separatorChar}${taskIdentity.name}"))
```

Comment:
will it work without throwing exceptions?

Reply:
Yes, the file is not created here, the object is just instantiated.